org-code-javabuilder/theater/src/main/java/org/code/theater/Theater.java

```diff
@@ -0,0 +1,181 @@
+package org.code.theater;
+
+public enum Instrument {
+   PIANO, BASS
+}
+
+public static class Theater {
```

Comment:
What is our reasoning for everything being static? The `Theater` has state (the series of images/sounds) and my intuition would be for it to have instance methods instead of static methods.

Reply:
another piece of state the theater could have is its height and width. Are we expecting students to get the height and width of the overall theater programmatically? Or are they just going to be told it's 400x400?

Reply:
The motivation for static methods is that we don't want students to have the ability to instantiate and use multiple theaters. We could go with a singleton pattern (in fact no matter what, behind the scenes I'd imagine this would be implemented with a singleton), but my reasoning was that would be another concept we'd need to explain.

Like Game Lab, the height is planned to be a fixed 400x400 pixels, so we can guarantee that.

Reply:
Makes sense, I agree it would be too much to try to explain a singleton to students. I do think it would be nice to have `getHeight` and `getWidth` methods on this class then

Reply:
Cool, I can add those!

Reply:
Would something like `Theater.getCanvas()` be more intuitive than the static methods?

Reply:
Then all operations are done against Canvas (and perhaps against a similar class for audio)

Reply:
Jamila believes that the static approach is the most sensible, let's go for that.

Reply:
Sorry to reopen this, but I feel very strongly that a class with getters and setters like this one shouldn't be static.  This will end up being very confusing for the student where these types of objects in Java are almost always instances.

As a compromise, what if we had a regular constructor, but it could only be called once?  Calling it a second time would throw an exception.  The student could then write `theater = new Theater()` to get a Theater, similar to Painter.

(I'm happy to chat with Jamila directly if that's easier.)

Reply:
Chatted with Jamila, she thinks this is good idea. While I did think this made the easiest to use API, it's using static methods to "hack" in global state, so yea, I think this is a worthwhile change.